zerver/tests/test_upload.py

```diff
@@ -47,7 +47,9 @@
 import requests
 import base64
 from datetime import timedelta
+from django.http import HttpRequest
 from django.utils.timezone import now as timezone_now
+from sendfile import _get_sendfile
```

Comment:
Is the upstream django-sendfile release good enough, or do we need a patched version?

Reply:
so upstream is good for nginx 1.5.9+ and unicode. I didn't happen to come across any issues with python 3 compatibility while testing till now.
We originally went to a patched version because we were trying to achieve 1.4.6. But later we released that unicode's were having issue because of django trying to keep x-accel-redirect param values in latin-1. If django didn't do that (which probably was complying with some standards we could probably have 1.4.6 as well)

Reply:
OK, makes sense.